new resource and data source : azurerm_oracle_cloud_vm_cluster (#27176)
hashicorp/terraform-provider-azurerm: website/docs/r/oracledatabase_cloud_vm_cluster.html.markdown

  resource_group_name = azurerm_resource_group.example.name
  cloud_exadata_infrastructure_name = azurerm_oracledatabase_exadata_infrastructure.example.name
}

resource "azurerm_oracledatabase_cloud_vm_cluster" "example" {
  name = "example-cloud-vm-cluster"
  resource_group_name = azurerm_resource_group.example.name
  location = azurerm_resource_group.example.location
  gi_version = "23.0.0.0"
  vnet_id = azurerm_virtual_network.example.id
  license_model = "BringYourOwnLicense"
  db_servers = [for obj in data.azurerm_oracledatabase_db_servers.example.db_servers : obj.ocid]
  ssh_public_keys = [ file("~/.ssh/id_rsa.pub") ]
  display_name = "example-cloud-vm-cluster"
  cloud_exadata_infrastructure_id = azurerm_oracledatabase_exadata_infrastructure.example.id
  cpu_core_count = 2
  hostname = "hostname"
  subnet_id = azurerm_subnet.example.id
}
```

## Arguments Reference

The following arguments are supported:

* `cloud_exadata_infrastructure_id` - (Required) The [OCID](https://docs.cloud.oracle.com/iaas/Content/General/Concepts/identifiers.htm) of the cloud Exadata infrastructure.

* `cpu_core_count` - (Required) The number of CPU cores enabled on the cloud VM cluster.

> * `db_servers` - (Required) The list of DB servers.

Reviewer: list of? hostnames/ ids? if host names
```suggestion
* `db_server_host_namess` - (Required) The list of DB server hostnames to use.
```

Author: @katbyte I agree with your comments on the naming of these arguments, however I have been asked to match the official OCI Terraform provider here (concerns about consistent customer experience), so I don't have a lot of freedom in changing them myself:
https://registry.terraform.io/providers/oracle/oci/latest/docs/data-sources/database_cloud_vm_cluster#db_servers

Reviewer 2: Typo `db_server_host_namess`

Author: These changes were not used, so the typo was not committed.

* `display_name` - (Required) The user-friendly name for the cloud VM cluster. The name does not need to be unique..

> * `gi_version` - (Required) A valid Oracle Grid Infrastructure (GI) software version.

Reviewer: we should avoid ambiguous acronyms
```suggestion
* `oracle_grid_nfrastructure_version` - (Required) A valid Oracle Grid Infrastructure (GI) software version.
```

Reviewer 2: Typo `oracle_grid_nfrastructure_version`.

Author: These changes were not used, so the typo was not committed.

* `hostname` - (Required) The hostname for the cloud VM cluster.

* `license_model` - (Required) The Oracle license model that applies to the cloud VM cluster, either `BringYourOwnLicense` or `LicenseIncluded`.

* `location` - (Required) The Azure Region where the Cloud VM Cluster should exist.

* `name` - (Required) The name which should be used for this Cloud VM Cluster.

* `resource_group_name` - (Required) The name of the Resource Group where the Cloud VM Cluster should exist.

* `ssh_public_keys` - (Required) The public key portion of one or more key pairs used for SSH access to the cloud VM cluster.

* `subnet_id` - (Required) The ID of the subnet associated with the cloud VM cluster.

* `vnet_id` - (Required) The ID of the vnet associated with the cloud VM cluster.

---

* `backup_subnet_cidr` - (Optional) The backup subnet CIDR of the vnet associated with the cloud VM cluster..

* `cluster_name` - (Optional) The cluster name for cloud VM cluster.

> * `data_collection_options` - (Optional) A `data_collection_options` block as defined below.

Reviewer: ```suggestion
* `data_collection` - (Optional) A `data_collection_options` block as defined below.
```
